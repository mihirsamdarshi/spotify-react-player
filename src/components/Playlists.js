import React from 'react';
import PlaylistBox from "./PlaylistBox";
import makeStyles from "@material-ui/core/styles/makeStyles";

const useStyles = makeStyles({
    cardPad: {
        marginBottom: 2000,
    }
})

const Playlists = props => {
    const classes = useStyles();
    const getIndividualItems = data => {
        console.log(data)
    };

    return (
        <div>
            {
                props.playlists.map(element =>
                    <PlaylistBox
                        name={element.name}
                        link={element.href}
                        numTracks={element.tracks.total}
                        img={element.images[0]}
                        className={classes.cardPad}
                    />
                )
            }
        </div>
    )
};

export default Playlists;

