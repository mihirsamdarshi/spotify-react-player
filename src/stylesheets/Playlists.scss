.media {
  width: 100%;
  margin-bottom: 10%;
}

.heightFix {
  max-height: 300px;
  text-align: center;
}

.card {
  display: flex;
  margin-bottom: 10px;
}

.card.playlist {
  flex-direction: column;
}

.card.selected {
  background-color: #007AFF;
}

.details {
  display: flex;
  flex-direction: column;
  background-color: #fff;
}

.content {
  flex: 1 0 auto;
};
