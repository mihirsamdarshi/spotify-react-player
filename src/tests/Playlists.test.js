import ReactDOM from 'react-dom';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import Playlists from '../components/Playlists';

const { act } = TestRenderer;

const fakePlaylistData = [
    {
        id: 123456789,
        name: 'Test',
        tracks: {
            total: 10,
        },
        images: [
            {
                url: 'test.com',
            },
        ],
    },
];

it('renders the Playlists component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Playlists playlists={fakePlaylistData} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should start with a playlist box', async () => {
    await act(async() => {
        const component = await TestRenderer.create(<Playlists playlists={fakePlaylistData}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
