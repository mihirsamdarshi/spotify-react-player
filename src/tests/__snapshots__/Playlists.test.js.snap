// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should start with a playlist box 1`] = `
<div
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root card playlist MuiPaper-rounded"
    onClick={[Function]}
  >
    <div
      className="heightFix"
    >
      <img
        className="media"
        src="test.com"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root details"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Test
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          10 tracks available
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
